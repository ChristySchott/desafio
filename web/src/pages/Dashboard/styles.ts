import styled from 'styled-components';

export const Main = styled.main`
  ul {
    list-style: none;
    display: flex;
    align-items: center;
    flex-direction: column;
    max-width: 1048px;
    margin: 10px auto 20px;
    position: relative;

    @media (min-width: 768px) {
      align-items: flex-start;
      flex-direction: row;
      flex-wrap: wrap;
      margin-top: 20px;
      padding: 0 15px;

      li {
        width: 33%;
      }
    }

    @media (min-width: 1440px) {
      padding: 0;
    }
  }
`;

export const Cart = styled.section`
  ul {
    list-style: none;
    display: flex;
    max-width: 1048px;
    margin: 10px auto 20px;

    li {
      background-position: center 60%;
    }

    @media (min-width: 768px) {
      margin-top: 20px;
      justify-content: flex-start;
      padding: 0 15px;

      li {
        width: 100%;
      }
    }

    @media (min-width: 1440px) {
      padding: 0;
    }
  }
`;

export const Title = styled.div`
  margin-bottom: 10px;
  max-width: 1048px;
  margin: 0 auto;
  padding: 15px 0 0 15px;

  @media (min-width: 768px) {
    margin-bottom: 20px;
  }

  a {
    color: #100c14;
  }

  h3 {
    color: #c61a28;
    text-transform: capitalize;
  }

  @media (min-width: 1440px) {
    padding: 15px 0 0;
    margin-top: 15px;
  }
`;
