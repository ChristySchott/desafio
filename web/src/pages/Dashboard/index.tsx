import React from 'react';

import { Main, Page } from './styles';

const Dashboard: React.FC = () => {
  return (
    <>
      <Page>
        <p>
          <a href="/">PÃ¡gina Inicial </a>
        </p>
      </Page>
      <Main>
        <p>Dashboard</p>
      </Main>
    </>
  );
};

export default Dashboard;
