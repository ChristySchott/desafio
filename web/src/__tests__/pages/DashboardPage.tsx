import React from 'react';
import { BrowserRouter } from 'react-router-dom';
import { render } from '@testing-library/react';

import Dashboard from '../../pages/Dashboard';

describe('Dashboard Page', () => {
  it('should render with no error', () => {
    const { container } = render(
      <BrowserRouter>
        <Dashboard />
      </BrowserRouter>,
    );

    expect(container).toBeTruthy();

    expect(container.firstChild).toMatchSnapshot();
  });
});
