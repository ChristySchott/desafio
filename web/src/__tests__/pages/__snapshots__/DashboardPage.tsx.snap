// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Dashboard Page should render with no error 1`] = `
<div
  class="sc-bdnylx"
>
  <div
    class="sc-gtssRu khYAvu"
  >
    <div>
      <p>
        <a
          href="/"
        >
          Acesse sua Conta
        </a>
        ou
        <a
          href="/"
        >
          Cadastre-se
        </a>
      </p>
    </div>
  </div>
  <div
    class="sc-dlnjPT eBLiYO"
  >
    <div
      class="sc-hKFyIo bkiLPl"
    >
      <div>
        <button
          class="sc-fujyUd hjIYgO"
        >
          <div />
          <div />
          <div />
        </button>
        <nav
          class="sc-pNWxx icIQWM"
        >
          <ul>
            <li>
              <a
                href="/"
              >
                Página inicial
              </a>
            </li>
            <li>
              <a
                href="/"
              >
                Camisetas
              </a>
            </li>
            <li>
              <a
                href="/"
              >
                Calças
              </a>
            </li>
            <li>
              <a
                href="/"
              >
                Sapatos
              </a>
            </li>
            <li>
              <a
                href="/"
              >
                Contato
              </a>
            </li>
          </ul>
        </nav>
      </div>
    </div>
    <div
      class="sc-jSFkmK dMcosL"
    >
      <img
        alt="Escrito Webjump com um símbolo de exclamação ao lado"
        src="logo.png"
      />
    </div>
    <div
      class="sc-gKAblj gRSOIu"
    >
      <svg
        color="#c81120"
        fill="currentColor"
        height="25"
        size="25"
        stroke="currentColor"
        stroke-width="0"
        style="color: rgb(200, 17, 32);"
        viewBox="0 0 24 24"
        width="25"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M10,18c1.846,0,3.543-0.635,4.897-1.688l4.396,4.396l1.414-1.414l-4.396-4.396C17.365,13.543,18,11.846,18,10 c0-4.411-3.589-8-8-8s-8,3.589-8,8S5.589,18,10,18z M10,4c3.309,0,6,2.691,6,6s-2.691,6-6,6s-6-2.691-6-6S6.691,4,10,4z"
        />
      </svg>
    </div>
    <div
      class="sc-iCoHVE iEnsXY"
    >
      <div
        class="sc-jrsJCI koRDCM"
      >
        <input />
        <button
          name="search"
          type="submit"
        >
          Buscar
        </button>
      </div>
    </div>
  </div>
  <div
    class="sc-eCApGN dVyjFc"
  >
    <ul>
      <li>
        <a
          href="/"
        >
          Página inicial
        </a>
      </li>
      <li>
        <a
          href="/"
        >
          Camisetas
        </a>
      </li>
      <li>
        <a
          href="/"
        >
          Calças
        </a>
      </li>
      <li>
        <a
          href="/"
        >
          Sapatos
        </a>
      </li>
      <li>
        <a
          href="/"
        >
          Contato
        </a>
      </li>
    </ul>
  </div>
</div>
`;
